bin/critterget.py

```diff
@@ -120,139 +173,178 @@ def apipost_with_response_code(uri, postdata=''):
+    # Print out in Splunk KV format.
+    # Return a dict with appId and appName.
 
-    # list of the attributes we'll hand to Apteligent, also the list we will create as Splunk Keys
-    summattrs = "appName,appType,crashPercent,dau,latency,latestAppStoreReleaseDate,latestVersionString,linkToAppStore,iconURL,mau,rating,role,appVersions"
-    atstring = "attributes=%s"%summattrs
-    gdata = apicall("apps", atstring)
+    params = {ATTRIBUTES: ','.join(SUMMARY_ATTRIBUTES)}
+    summary_data = apicall(APPS, params)
 
     AppDict = {}
-    for appId in gdata.keys():
-        printstring = u'{} MessageType="AppSummary" appId={} '.format(DATETIME_OF_RUN, appId)
-        slist = summattrs.split(",")
-        for atname in slist:
-            printstring += u'{}="{}"'.format(atname,gdata[appId][atname])
-            if atname == "appName" : AppDict[appId]= {'name': gdata[appId][atname], 'versions': gdata[appId]['appVersions']}
+    for appId in summary_data[DATA].keys():
+        printstring = u'{} MessageType="AppSummary" appId={} '.format(
+            DATETIME_OF_RUN, appId
+        )
+
+        for attribute_name in SUMMARY_ATTRIBUTES:
+            printstring += u'{}="{}"'.format(
+                attribute_name,
+                summary_data[DATA][appId][attribute_name]
+            )
+            if attribute_name == APPNAME:
+                AppDict[appId] = {
+                    NAME: summary_data[DATA][appId][attribute_name],
+                    VERSIONS: summary_data[DATA][appId][APPVERSIONS]
+                }
         print printstring
 
     return AppDict
 
 
 def getCrashSummary(appId, appName):
-# given an appID and an appName, produce a Splunk KV summary of the crashes for a given timeframe
+    # given an appID and an appName, produce a Splunk KV summary of the
+    # crashes for a given timeframe
+
     mystime = scopetime()
     crashdata = None
-    crashattrs = "hash,lastOccurred,sessionCount,uniqueSessionCount,reason,status,displayReason,name"
 
     http_code = None
     retry = 0
     while http_code != 200:
-        http_code, crashdata = apicall_with_response_code("app/%s/crash/summaries" % appId, "lastOccurredStart=%s" % mystime)
+        http_code, crashdata = apicall_with_response_code(
+            "app/crash/summaries/{}".format(
+                appId,
+                {'lastOccurredStart': mystime}
+            )
+        )
         retry += 1
         if retry > MAX_RETRY:
             break
 
     CrashDict = {}
     if http_code != 200:
-        print u'{} MessageType="CrashSummary" appId={} appName="{}" ' \
-              u'Could not get crash summaries for {}. ' \
-              u'Code: {} after retry {}'.format(DATETIME_OF_RUN,
+        print (u'{} MessageType="CrashSummary" appId={} appName="{}" '
+              u'Could not get crash summaries for {}. '
+              u'Code: {} after retry {}').format(DATETIME_OF_RUN,
                                                 appId, appName,
```

Comment:
This one doesn't line up either.  Also strange we're doing one var per line except this one.